### ISSUE TITLE
Output difference anomaly under equivalent migration in logdet operator

### ISSUE THREAD

**Issue by whitephilomel** [pytorch/pytorch#176415]
### 🐛 Describe the bug

We observed a severe NaN-masking inconsistency between `torch.logdet` and TensorFlow when processing 4D batched matrices (`shape=[4, 4, 2, 2]`, `float32`).

For a batch containing 9 matrices with non-positive determinants, PyTorch handles the valid matrices, computing float values while naturally returning `NaN` only for the non-positive boundaries. TensorFlow, however, returns significantly more `NaN` values across the batch tensor, leading to an output mismatch where the maximum difference is `nan`. This discrepancy in handling batched LU factorization vs Cholesky decomposition creates unpredictable numerical mappings in translation pipelines.

```python
import numpy as np
import torch
import tensorflow as tf

np.random.seed(42)
input_np = np.random.randn(4, 4, 2, 2).astype(np.float32)

input_pt = torch.tensor(input_np)
out_pt = torch.logdet(input_pt)

input_tf = tf.constant(input_np)
out_tf = tf.linalg.logdet(input_tf)

pt_np_out = out_pt.numpy()
tf_np_out = out_tf.numpy()
max_diff = np.max(np.abs(pt_np_out - tf_np_out))
neg_det_count = np.sum(np.linalg.det(input_np) <= 0)
print(f"Number of matrices with non-positive determinant: {neg_det_count}")
print(f"PyTorch logdet output: {pt_np_out}")
print(f"TensorFlow logdet output: {tf_np_out}")
print(f"Maximum difference: {max_diff}")
```

output:

```
Number of matrices with non-positive determinant: 9
PyTorch logdet output: [[-0.16716182 -1.6604435  -0.7548522          nan]
 [ 0.5398615          nan         nan         nan]
 [-0.12048236         nan         nan         nan]
 [-0.8231379          nan         nan -1.0002422 ]]
TensorFlow logdet output: [[-1.0876427        nan        nan        nan]
 [       nan        nan        nan        nan]
 [       nan        nan        nan        nan]
 [       nan        nan        nan        nan]]
Maximum difference: nan
```

### Versions

Collecting environment information...
PyTorch version: 2.7.1+cpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 11 家庭中文版 (10.0.26100 64 位)
GCC version: (x86_64-win32-seh-rev0, Built by MinGW-W64 project) 8.1.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: N/A

Python version: 3.10.18 | packaged by Anaconda, Inc. | (main, Jun  5 2025, 13:08:55) [MSC v.1929 64 
bit (AMD64)] (64-bit runtime)                                                                       Python platform: Windows-10-10.0.26100-SP0
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

CPU:
Name: AMD Ryzen 7 8845H w/ Radeon 780M Graphics
Manufacturer: AuthenticAMD
Family: 107
Architecture: 9
ProcessorType: 3
DeviceID: CPU0
CurrentClockSpeed: 3801
MaxClockSpeed: 3801
L2CacheSize: 8192
L2CacheSpeed: None
Revision: 29954

Versions of relevant libraries:
[pip3] numpy==1.26.4
[pip3] onnx==1.19.0
[pip3] onnx_graphsurgeon==0.5.8
[pip3] onnx2tf==1.28.8
[pip3] onnxruntime==1.22.1
[pip3] optree==0.16.0
[pip3] torch==2.7.1
[pip3] torchaudio==2.7.1
[pip3] torchvision==0.22.1
[conda] numpy                     1.26.4                   pypi_0    pypi
[conda] optree                    0.16.0                   pypi_0    pypi
[conda] torch                     2.7.1                    pypi_0    pypi
[conda] torchaudio                2.7.1                    pypi_0    pypi
[conda] torchvision               0.22.1                   pypi_0    pypi

cc @peterjc123 @mszhanyi @skyline75489 @nbcsm @iremyux @Blackhex @jianyuh @nikitaved @mruberry @walterddr @xwang233 @Lezcano
